mmv1/third_party/terraform/services/storage/resource_storage_folder_test.go

```diff
@@ -0,0 +1,249 @@
+			},
+			{
+				Config: testAccStorageFolder_storageBucketAndFolders(bucketName, true, true),
+			},
+		},
+	})
+}
+
+func testAccStorageFolder_storageBucket(bucketName string, hnsFlag bool, forceDestroy bool) string {
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "bucket" {
+  name                        = "%s"
+  location                    = "EU"
+  uniform_bucket_level_access = true
+  hierarchical_namespace {
+	enabled = %t
+  }
+  force_destroy = %t
+}
+`, bucketName, hnsFlag, forceDestroy)
+}
+
+func testAccStorageFolder_storageBucketObject(bucketName string, hnsFlag bool, forceDestroy bool, fileName string) string {
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "bucket" {
+  name                        = "%s"
+  location                    = "EU"
+  uniform_bucket_level_access = true
+  hierarchical_namespace {
+	enabled = %t
+  }
+  force_destroy = true
+}
+resource "google_storage_folder" "folder" {
+  bucket        = google_storage_bucket.bucket.name
+  name          = "folder/"
+  force_destroy = %t
+}
+resource "google_storage_folder" "subfolder" {
+  bucket        = google_storage_bucket.bucket.name
+  name          = "${google_storage_folder.folder.name}subfolder/"
+  force_destroy = %t
+}  
+resource "google_storage_bucket_object" "object" {
+  name   = "${google_storage_folder.subfolder.name}tffile"
+  bucket = google_storage_bucket.bucket.name
+  source = "%s"
+}  
+`, bucketName, hnsFlag, forceDestroy, forceDestroy, fileName)
+}
+
+func testAccStorageFolder_storageFolder(forceDestroy bool) string {
+	return fmt.Sprintf(`
+resource "google_storage_folder" "folder" {
+  bucket        = google_storage_bucket.bucket.name
+  name          = "folder/"
+  force_destroy = %t
+}
+resource "google_storage_folder" "subfolder" {
+  bucket        = google_storage_bucket.bucket.name
+  name          = "${google_storage_folder.folder.name}name/"
+  force_destroy = %t
+}  
+`, forceDestroy, forceDestroy)
+}
+
+func testAccStorageFolder_storageOneFolder(forceDestroy bool) string {
+	return fmt.Sprintf(`
+resource "google_storage_folder" "folder" {
+  bucket        = google_storage_bucket.bucket.name
+  name          = "folder/"
+  force_destroy = %t
+} 
+`, forceDestroy)
+}
+
+func testAccStorageFolder_storageBucketAndFolders(bucketName string, forceDestroy bool, subFolderForceDestroy bool) string {
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "bucket" {
+  name                        = "%s"
+  location                    = "EU"
+  uniform_bucket_level_access = true
+  hierarchical_namespace {
+	enabled = "true"
+  }
+}
+resource "google_storage_folder" "folder" {
+  bucket        = google_storage_bucket.bucket.name
+  name          = "folder/"
+  force_destroy = %t
+}
+resource "google_storage_folder" "subfolder" {
+  bucket        = google_storage_bucket.bucket.name
+  name          = "${google_storage_folder.folder.name}subfolder/"
+  force_destroy = %t
```

Comment:
Nit: it would be nice to remove this line, as I don't think it has any influence on the behavior, and actually implies that the user would need to set `force_destroy` in all subfolders for deletion to work.

Reply:
updated